src/wallet/wallet.cpp

```diff
@@ -2106,7 +2106,7 @@ void CWallet::AvailableCoins(vector<COutput> &vCoins, bool fOnlyConfirmed, const
                 bool fSpendableIn = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) || (coinControl && coinControl->fAllowWatchOnly && (mine & ISMINE_WATCH_SOLVABLE) != ISMINE_NO);
                 bool fSolvableIn = (mine & (ISMINE_SPENDABLE | ISMINE_WATCH_SOLVABLE)) != ISMINE_NO;
 
-                vCoins.push_back(COutput(pcoin, i, nDepth, fSpendableIn, fSolvableIn));
+                vCoins.emplace_back(COutput(pcoin, i, nDepth, fSpendableIn, fSolvableIn));
```

Comment:
`vCoins.emplace_back(COutput(pcoin, i, nDepth, fSpendableIn, fSolvableIn));`

This is a good place to show why you can't just simply find and replace all instances of `push_back` with `emplace back`. This is still allocating an entire temporary COutput object **A** to pass to `emplace_back`. Another COutput object is allocated for vCoins itself, moving/copying the contents of the temporary **A** object to itself. This is pretty much the same thing that push_back is doing so this solves nothing

However, consider the following instead:
`
vCoins.emplace_back(pcoin, i, nDepth, fSpendableIn, fSolvableIn);
`

Notice how even visually you're not allocating a temporary object to pass to `emplace_back`? This is where `emplace_back` begins to shine, but you will have to handle these changes on a case by case basis to make sure you're actually reducing allocations and not just adding more work

Here's a snippet of code you can play around with to see it in action: http://cpp.sh/2vzu5